src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -241,9 +312,25 @@ private ArmClientExtensions EnsureArmClientExtensions()
 
         private Dictionary<Schema, TypeProvider>? _models;
 
-        public Dictionary<Schema, TypeProvider> ResourceSchemaMap => _resourceModels ??= BuildResourceModels();
+        public Dictionary<Schema, TypeProvider> ResourceSchemaMap
+        {
+            get
+            {
+                if (_resourceModels == null)
```

Comment:
Do you need these properties to be lazy? Can you initialize them directly?

Reply:
@AlexanderSher I tried to initialize them in constructor, and failed... There is a cyclic reference between `MgmtOutputLibrary<BuildContext>` and `BuildContext`. A significant refactoring is required so that we initialize the models in constructor.